awive/config.py

```
@@ -95,7 +96,45 @@ def di(i: int, j: int):
         x[:, 0] *= -1
         return x.tolist()
 
+    def convert_str_keys_to_tuples(
+        self, input_dict: dict[str, float]
+    ) -> dict[tuple[int, int], float]:
+        """Convert string-represented tuple keys to actual tuples"""
+        result = {}
+        for key, value in input_dict.items():
+            if (
+                isinstance(key, str)
+                and key.startswith("(")
+                and key.endswith(")")
+            ):
+                try:
+                    # Safely evaluate the string as a tuple
+                    tuple_key = literal_eval(key)
+                    if isinstance(tuple_key, tuple):
+                        result[tuple_key] = value
+                    else:
+                        raise ValueError(f"Key '{key}' is not a tuple")
+                except (ValueError, SyntaxError) as e:
+                    raise ValueError(
+                        f"Invalid tuple format in key '{key}'"
+                    ) from e
+            else:
+                # Keep the key as is if it's not a string-represented tuple
+                result[key] = value
```

If the keys are not strings representing valid tuples, this parser should raise a ValueError exception. Don’t you think? Or is there another reason to keep these keys?

---

Yeah, I think you are right. There's no reason to keep the key. It should raise a ValueError exception

---

About the type hint dict[tuple[int, int], float]. I see that is only useful when parsing config.py as json dict and not the actual yaml format we are using. I think there are other forms of format that would allow tuples as keys. Since I don't believe we would change the data format on the config files, I think i'd be easier to have the type hint dict[str, float]

---

Ok, keep it as dict[str, float]